tests/unit/tbmodule.test.js

```diff
@@ -0,0 +1,279 @@
+            });
+            it('mixes-in `TBLog`', () => {
+                expect(window.TBLog).toHaveBeenCalledWith(module);
+            });
+
+            describe('creates default properties', () => {
+                describe('the `shortname` property', () => {
+                    it('strips spaces when derived from `name`', () => {
+                        expect(module.shortname).toStrictEqual('TestModuleName');
+                    });
+                    it('does not strip spaces when explicitly set', () => {
+                        const shortname = 'another test';
+                        module.shortname = shortname;
+                        expect(module.shortname).toStrictEqual(shortname);
+                    });
+                });
+                describe('the `config` property', () => {
+                    it('is an object', () => {
+                        expect(module.config).toBeInstanceOf(Object);
+                    });
+                    it('defaults property `betamode` to `false`', () => {
+                        expect(module.config.betamode).toBe(false);
+                    });
+                    it('defaults property `devmode` to `false`', () => {
+                        expect(module.config.devmode).toBe(false);
+                    });
+                });
+                describe('the `settings` property', () => {
+                    it('is an object', () => {
+                        expect(module.settings).toBeInstanceOf(Object);
+                    });
+                    it('creates a property `enabled` that is an object', () => {
+                        expect(module.settings.enabled).toBeInstanceOf(Object);
+                    });
+                    it('defaults property `type` to "boolean"', () => {
+                        expect(module.settings.enabled.type).toBe('boolean');
+                    });
+                    it('defaults property `default` to `false`', () => {
+                        expect(module.settings.enabled.default).toBe(false);
+                    });
+                    it('defaults property `betamode` to `false`', () => {
+                        expect(module.settings.enabled.betamode).toBe(false);
+                    });
+                    it('defaults property `hidden` to `false`', () => {
+                        expect(module.settings.enabled.hidden).toBe(false);
+                    });
+                    it('defaults property `title` to "Enable $name"', () => {
+                        expect(module.settings.enabled.title).toBe(`Enable ${moduleName}`);
+                    });
+                });
+                it('the `settingsList` property is an array with a single string element "enabled"', () => {
+                    expect(module.settingsList).toStrictEqual(['enabled']);
+                });
+            });
+        });
+
+        describe('instance methods', () => {
+            describe('the `register_setting()` method', () => {
```

Comment:
It's a bit too verbose IMO. We can go with this:
```suggestion
            describe('register_setting()', () => {
```

Only thing that matters really is being verbose in the `it('adds foo')` stuff. You agree @creesch?